hibernate-core/src/main/java/org/hibernate/metamodel/model/domain/package-info.java

```diff
@@ -16,4 +16,7 @@
  *
  * @see jakarta.persistence.metamodel
  */
+@Incubating
 package org.hibernate.metamodel.model.domain;
```

Comment:
This could be a problem for Hibernate Search, as we had to move to these types (`EntityDomainType` in particular) in order to avoid deprecated/internal types (entity persisters in particular). So we don't really have a stable alternative to these contracts and may be affected by changes, unexpectedly breaking compatibility.

Reply:
Well the package that contains this package was already marked `@Incubating`. Now, I guess it's unclear whether `@Incubating` is supposed to be inherited by subpackages, but in this case I think it's simply an acknowledgement of reality that these contracts are pretty new and we can't guarantee that they won't change..

Reply:
[Of course, we're not going to do things that arbitrarily break Search without checking with you first.]

Reply:
Right, and I'm fine with stuff being broken as well, if we have solutions in Hibernate Search. Just pointing out that this may result in me reporting breaking changes, which I know can be irritating for the ORM team (because you can't know in advance what change can possibly break Hibernate Search, and I can't possibly add tests that simulate all of Hibernate Search's API/SPI uses, at least not if those need to be always in sync with Hibernate Search).

As long as everyone involved  agrees with that, `@Incubating` is fine :)